akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -26,13 +26,26 @@ sealed abstract class OutPort
  * is the InPort (which does not bear an element type because Modules only
  * express the internal structural hierarchy of stream topologies).
  */
-final class Inlet[-T](override val toString: String) extends InPort
+object Inlet {
+  def apply[T](toString: String): Inlet[T] = new Inlet[T](toString)
+}
+
+final class Inlet[-T] private (override val toString: String) extends InPort {
+  def carbonCopy[R](): Inlet[R] = Inlet(toString)
```

Comment:
I think this encoding (instead of manual copying using `new` and passing in the correct name) and it meshes very well with other classes' carbonCopy methods.